### ISSUE TITLE
Circulars: Convert emails to hyperlinks

### ISSUE THREAD

**Issue by tylerbarna** [nasa-gcn/gcn.nasa.gov#2456]
# Description
We currently have functionality that automatically inserts hyperlinks for URLs, DOIs, TNSs, and GCN Circulars when they're referenced in Circulars. It may be beneficial to automatically hyperlink email addresses, both in the body of Circulars and in the submitter information. 

The submitter email will be a more straightforward regular expression, as it can simply be a regular expression that matches any string enclosed by `<>`, but the regular expression to link to emails in a Circular may be less straightforward.

# Acceptance criteria

- [ ] Implement a regular expression (or regular expressions) that match any strings that contain an email `address@provider.TLD` and convert them to a hyperlink in the format `mailto:address@provider.TLD`
- [ ] Ensure arbitrary strings that contain the `@` symbol are not matched by the regular expression
- [ ] Apply this regular expression to both the submitter email and any potential emails in the body of Circulars
- [ ] The function that evaluates the regular expression should be able to handle instances where the submitter does not contain an email without raising an error

**Comment by lpsinger** [comment]
It already works for Markdown-formatted Circulars because this is done by the autolink literals plugin.

**Comment by lpsinger** [comment]
Here's where the code lives: https://github.com/nasa-gcn/gcn.nasa.gov/blob/main/app/routes/circulars.%24circularId.(%24version)/rehypeAutolinkLiteral.ts

**Comment by tylerbarna** [comment]
> It already works for Markdown-formatted Circulars because this is done by the autolink literals plugin.

does this include the submitter email in the header?

**Comment by lpsinger** [comment]
> > It already works for Markdown-formatted Circulars because this is done by the autolink literals plugin.
> 
> does this include the submitter email in the header?

No, but you could easily process that using remark/rehype.

**Comment by tylerbarna** [comment]
@lpsinger that's what I was thinking, should this issue be migrated to [remark-rehype-astro](https://github.com/nasa-gcn/remark-rehype-astro)?

**Comment by lpsinger** [comment]
No. As I said, this code is in https://github.com/nasa-gcn/gcn.nasa.gov/blob/main/app/routes/circulars.%24circularId.(%24version)/rehypeAutolinkLiteral.ts.

**Comment by lpsinger** [comment]
Here is an example of a Circular containing an email address: https://gcn.nasa.gov/circulars/22081

It is not captured as a link. However, if one switches it to Markdown mode, it is. So this is working for Markdown but not plain text at the moment.

**Comment by tylerbarna** [comment]
@lpsinger Are there characters that aren't rendering correctly for you in that circular? This is what I'm seeing on my Mac (on Chrome):

<img width="758" alt="Screenshot 2024-12-08 at 14 23 23" src="https://github.com/user-attachments/assets/20c41176-7753-4f76-8ed6-c5aeaa4674a9">

**Comment by lpsinger** [comment]
> @lpsinger Are there characters that aren't rendering correctly for you in that circular? This is what I'm seeing on my Mac (on Chrome):

Yes. That's a known issue. GCN Classic couldn't handle non-ASCII characters consistently. We have a lot of mangled text from the legacy system.
